internal/provider/network_connection_data_source.go

```diff
@@ -0,0 +1,66 @@
+package provider
+
+import (
+	"context"
+	"fmt"
+
+	dcsdk "github.com/doublecloud/go-sdk"
+	dcgennet "github.com/doublecloud/go-sdk/gen/network"
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+)
+
```

Comment:
Could you add?
```golang
// Ensure provider defined types fully satisfy framework interfaces.
// Ensure provider defined types fully satisfy framework interfaces.
var _ datasource.DataSource = &NetworkConnectionDataSource{}
```

Reply:
Why? It is already checking in compilation here:
```go
func NewNetworkConnectionResource() resource.Resource {
	return &NetworkConnectionResource{}
}
```

Reply:
Just a little helper for your IDE, to mark some unimplemented methods before compilation.